fastlane_core/lib/fastlane_core/cert_checker.rb

```diff
@@ -144,29 +144,71 @@ def self.install_missing_wwdr_certificates
       # Install all Worldwide Developer Relations Intermediate Certificates listed here: https://www.apple.com/certificateauthority/
       missing = WWDRCA_CERTIFICATES.map { |c| c[:alias] } - installed_wwdr_certificates
       missing.each do |cert_alias|
-        install_wwdr_certificate(cert_alias)
+        url = WWDRCA_CERTIFICATES.find { |c| c[:alias] == cert_alias }.fetch(:url)
+        file = Tempfile.new(File.basename(url))
+
+        unless fetch_certificate(url, file)
+          UI.verbose("Could not fetch certificate #{cert_alias}")
+          next
+        end
+
+        unless check_expiry(filename)
```

Comment:
I might be missing something, but I don't see where `filename` is declared? Or maybe you meant to use `file.path` here?

> Quick ruby question: When creating a Tempfile object, is that then getting cleared as soon as it is out of scope. F.ex., if I create a Tempfile in the fetch method, after fetching the certificate, I can't just pass that Tempfile to the next method?

Great question! I agree we should be more careful as to when the Tempfile gets deleted.

I suggest you to use `Tempfile.create { |file| … }` with a block to wrap the 3 calls to `fetch_certificate`, `check_expiry` and `import_wwdr_certificate`, which guarantees that the temp file is properly cleaned up and deleted when we exit the scope (and this even if we exit early with a `next` or an exception occurs)—as opposed to `Tempfile.new` which doesn't (more info [here](https://www.rubyguides.com/2019/05/ruby-tempfile/)).

So something like:

```ruby
url = WWDRCA_CERTIFICATES.find { |c| c[:alias] == cert_alias }.fetch(:url)
Tempfile.create('fastlane-match-wwdr-cert-') do |tmpfile|
  unless fetch_certificate(url, tmpfile.path)
    UI.verbose("Could not fetch certificate #{cert_alias}")
    next
  end

  unless check_expiry(tmpfile.path)
    UI.verbose("Skipping installation of certificate: #{filename} because it is invalid.") 
    next
  end

  UI.message("Installing WWDR certificate '#{cert_alias}'")
  import_wwdr_certificate(tmpfile.path)
end
```

Reply:
Ah.. stupid leftover (the filename thingy.. I'll clear that up)

Good suggestion regarding the Tempfile. I'll also dive into that. Thanks for the info link! Much appreciated 🙌

Reply:
Great suggestion indeed! I didn't know of this Tempfile.create with block 🤩